# Indexing reference genome using Subread
```shell
# Requesting computation resources:
sinteractive -c 28 -t 00:05:00 -J Indexing -A PAS2658
cd /fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch

# Execute R Interactively:
module load R/4.3.0-gnu11.2
R

# Setting working directory
setwd("/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch")

# Build Index
library(Rsubread)
buildindex(basename="P_fructicola_Index",reference="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Reference/Sooty_Blotch/P_fructicola.fasta")

# quit R
quit("no")

# Checking output
ls
```

# Setting up alignment
```shell
# First we will prepare a R script for this alignment and counting of features, look at the parttern and structure carefully (this script will need to be modified in the editor before execution):
cd /fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch
tee -a SubreadAlign.R <<EOF
library(Rsubread)

# Setting working directory
setwd("/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch")

# Aligning RNA-seq data to P. fructicola reference using Rsubread
# for PD_05d
align(index="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch/P_fructicola_Index",readfile1="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PD_05d_R1.fastq",readfile2="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PD_05d_R2.fastq",input_format="FASTQ",output_format="BAM",output_file="PD_05d.bam",nsubreads=14,maxMismatches=5,nthreads=20,phredOffset=33,unique=TRUE,minFragLength=50,maxFragLength=200,PE_orientation="fr",nTrim5=0,nTrim3=0,readGroupID=NULL,readGroup=NULL,color2base=FALSE,DP_GapOpenPenalty=-1, DP_GapExtPenalty=0,DP_MismatchPenalty=0,DP_MatchScore=2)

# for PD_15d
align(index="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch/P_fructicola_Index",readfile1="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PD_15d_R1.fastq",readfile2="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PD_15d_R2.fastq",input_format="FASTQ",output_format="BAM",output_file="PD_15d.bam",nsubreads=14,maxMismatches=5,nthreads=20,phredOffset=33,unique=TRUE,minFragLength=50,maxFragLength=200,PE_orientation="fr",nTrim5=0,nTrim3=0,readGroupID=NULL,readGroup=NULL,color2base=FALSE,DP_GapOpenPenalty=-1, DP_GapExtPenalty=0,DP_MismatchPenalty=0,DP_MatchScore=2)

# for PEG15d
align(index="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch/P_fructicola_Index",readfile1="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PEG15d_R1.fastq",readfile2="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Raw_Data/Sooty_Blotch/PEG15d_R2.fastq",input_format="FASTQ",output_format="BAM",output_file="PEG15d.bam",nsubreads=14,maxMismatches=5,nthreads=20,phredOffset=33,unique=TRUE,minFragLength=50,maxFragLength=200,PE_orientation="fr",nTrim5=0,nTrim3=0,readGroupID=NULL,readGroup=NULL,color2base=FALSE,DP_GapOpenPenalty=-1, DP_GapExtPenalty=0,DP_MismatchPenalty=0,DP_MatchScore=2)

# Counting features
fc <- featureCounts(files=c("PD_05d.bam","PD_15d.bam","PEG15d.bam"),annot.ext="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Reference/Sooty_Blotch/P_fructicola.gff3",isGTFAnnotationFile=TRUE,GTF.featureType="gene",GTF.attrType="ID",useMetaFeatures=TRUE,countMultiMappingReads=TRUE,fraction=TRUE,genome="/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Reference/Sooty_Blotch/P_fructicola.fasta",isPairedEnd=TRUE,autosort=TRUE,nthreads=20)

# Prepare a summary table with relevant information
setwd("/fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch")
write.table(x=data.frame(fc[["annotation"]][,c("GeneID","Length")],fc[["counts"]],stringsAsFactors=FALSE),file="PD05d_PD15d_PEG15d_total_counts.txt",quote=FALSE,sep="\t",row.names=FALSE)
stat=data.frame(fc[["stat"]])
write.csv(stat,"PD05d_PD15d_PEG15d_total_counts.csv")

# Quit R
quit(save = "no")

EOF
```
Note: Add the $ symbols where apprpiate in the R script by using the editor!
Then, we make a script to submit in the batch to execute this R routine
```shell
tee -a SubreadAlign.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Alignment_Counting
#SBATCH --time=01:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2303

# Working directory
cd /fs/scratch/PASXXXX/YourOSC_ID/Gene_Expression/Rsubread/Sooty_Blotch

# load R
module load R/4.3.0-gnu11.2

# Run script in batch mode:
R CMD BATCH --no-save SubreadAlign.R
EOF
```
And submit the job to the scheduler:
```shell
sbatch SubreadAlign.sh
squeue -u osu8618
```
